microservices-connector/config/authN-authZ/README.md

```diff
@@ -156,6 +171,123 @@ curl -X POST $accessUrl -d '{"text":"What is the revenue of Nike in 2023?","para
+
+**Installation and configuration of oauth2-proxy and OIDC Provider**
+
+Here we take Keycloak as the sample OIDC Provider to use in the example. Follow the steps to install and configure Keycloak.
+
+```bash
+cd $(pwd)
+kubectl apply -f $(pwd)/config/authN-authZ/keycloak_install.yaml
+
+# get the ip and port to access keycloak.
+export HOST_IP=$(kubectl config view --minify -o jsonpath='{.clusters[0].cluster.server}' | cut -d '/' -f3 | cut -d ':' -f1)
+export KEYCLOAK_PORT=$(kubectl get svc keycloak -o jsonpath='{.spec.ports[0].nodePort}')
+export KEYCLOAK_ADDR=${HOST_IP}:${KEYCLOAK_PORT}
+```
+
+**Note:** Double check if the host ip captured is the correct ip.
+
+Access the Keycloak admin console through the `KEYCLOAK_ADDR` to configure the users as well as roles. Use the username and password specified in the yaml file to login.
+
+The user management is done via Keycloak and the configuration steps look like this:
+
+1. Create a new realm named `chatqna` within Keycloak.
+
+<img src="./docs/create_realm.png" width="600" height="300">
+
+2. Create a new client called `chatqna` and set `Client authentication` to 'On'. Set "http://chatqna-ui.com:${INGRESS_PORT}/*" in the `Valid redirect URIs` part. Under the Credentials tab you will now be able to locate `<your client's secret>`, which will be used in the oauth2-proxy configs.
+
+<img src="./docs/create_client_1.png" width="600" height="300">
+
+<img src="./docs/create_client_2.png" width="600" height="300">
+
+<img src="./docs/create_client_3.png" width="600" height="300">
+
+3. Access the dedicated mappers pane by clicking `<your client's id>-dedicated`, located under Assigned client scope to configure a new `Audience` mapper with name `aud-mapper-<your client's id>`. And include Audience in your client with `ID token` and `access token` set to `On`.
+
+<img src="./docs/add_mapper.png" width="600" height="300">
+
+4. Create new roles `user` and `viewer` by navigating to `<your client's id> -> Roles`.
+
+5. Create a new user name as `mary` and another user as `bob` with `Email verified` set to `On`. Set passwords for both users (set 'Temporary' to 'Off').
+
+<img src="./docs/create_user.png" width="600" height="300">
+
+5. Create a new Client Scope with the name `groups` in Keycloak with `Include in Token Scope` set as `On`. Include a mapper of type `Group Membership` and set the `Token Claim Name` to `groups`. If the "Full group path" option is selected, you need to include a "/" separator in the group names defined in the --allowed-group option of OAuth2 Proxy. Example: "/groupname". After creating the Client Scope named `groups` you will need to attach it to your client. Go to Clients and find `<your client's id> -> Client scopes` and add client scope and select `groups` and choose `Optional` and you should now have a client that maps group memberships into the JWT tokens so that Oauth2 Proxy may evaluate them.
```

Comment:
```suggestion
6. Create a new Client Scope with the name `groups` in Keycloak with `Include in Token Scope` set as `On`. Include a mapper of type `Group Membership` and set the `Token Claim Name` to `groups`. If the "Full group path" option is selected, you need to include a "/" separator in the group names defined in the --allowed-group option of OAuth2 Proxy. Example: "/groupname". After creating the Client Scope named `groups` you will need to attach it to your client. Go to Clients and find `<your client's id> -> Client scopes` and add client scope and select `groups` and choose `Optional` and you should now have a client that maps group memberships into the JWT tokens so that Oauth2 Proxy may evaluate them.
```

Reply:
Done